### PR TITLE
PD-53 Commands implementation

### PR DESCRIPTION
**Pull request by crapper**

### CODE REVIEW

File: src/command/Commands.ts [hunk 1]
```diff
@@ -499,3 +512,338 @@ export class CommandLineSpec implements InputSpec<typeof StringT> {
...
+      buildInputSpecByUsages([
+        {
+          name: "source_index",
+          paramType: NumberT,
+          description: "the index of the source field",
+          check: param => {
+            if (param.isDouble()) return fail("Source index must be a integer.");
+            if (param.getInt() < 0) return fail("Source index must be a positive integer or zero.");
+            const { app } = getRootStore();
+            if (param.getInt() >= app.diagram.size()) return fail("Index out of range.");
```
Comment (Jerrylum) [inline src/command/Commands.ts:664]: Change to `Source index out of range.`

File: src/command/Commands.ts [hunk 2]
```diff
@@ -499,3 +512,338 @@ export class CommandLineSpec implements InputSpec<typeof StringT> {
...
+        },
+        {
+          name: "destination_index",
+          paramType: NumberT,
+          description: "the index of the destination field",
+          check: param => {
+            if (param.isDouble()) return fail("Destination index must be a integer.");
+            if (param.getInt() < 0) return fail("Destination index must be a positive integer or zero.");
+            const { app } = getRootStore();
+            if (param.getInt() >= app.diagram.size()) return fail("Index out of range.");
```
Comment (Jerrylum) [inline src/command/Commands.ts:676]: Change to `Destination index out of range.`
Make it clear

File: src/command/Commands.ts [hunk 3]
```diff
@@ -99,7 +99,19 @@ export abstract class Command {
   abstract handle(params: Parameter<ParameterType>[]): HandleResult;
 
   static getAvailableCommands(): Command[] {
-    return [new UndoCommand(), new RedoCommand(), new AddCommand(), new ConfigCommand(), new HelpCommand()];
+    return [
```
Comment (Jerrylum) [inline src/command/Commands.ts:102]: Reorder the list of commands from A to Z

File: src/command/Commands.ts [hunk 4]
```diff
@@ -499,3 +512,338 @@ export class CommandLineSpec implements InputSpec<typeof StringT> {
 export function buildInputSpecByCommands(commands: Command[]): InputSpec<typeof StringT> | null {
   return new CommandLineSpec(commands);
 }
+
```
Comment (Jerrylum) [inline src/command/Commands.ts:515]: Define command classes before all the build functions

**Review by Jerrylum — CHANGES_REQUESTED** [review]

File: src/command/Commands.ts (hunk 3 above)
Comment (crapper) [reply src/command/Commands.ts:102]: Updated.

File: src/command/Commands.ts (hunk 2 above)
Comment (crapper) [reply src/command/Commands.ts:676]: Updated.

File: src/command/Commands.ts (hunk 4 above)
Comment (crapper) [reply src/command/Commands.ts:515]: Moved.

File: src/command/Commands.ts (hunk 1 above)
Comment (crapper) [reply src/command/Commands.ts:664]: Updated.

**Review by Jerrylum — APPROVED** [review]
LGTM

**Pull request merged**